api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiAuthenticator.java

```diff
@@ -0,0 +1,28 @@
+package org.openmrs.module.registrationcore.api.mpi.common;
+
+/**
+ * This service perform authentication to MPI server. It is a Spring managed bean which is configured
+ * in moduleApplicationContext.xml.
+ */
+public interface MpiAuthenticator {
```

Comment:
@djazayeri I think that obviously everyone MPI implementation will provide authentication by username/password credentials. And more, probably someone of them will provide alternative authentication ways. So, this Interface is important.